lib/rubydoop/dsl.rb

```diff
@@ -425,12 +428,16 @@ def wait_for_completion(verbose)
       end
 
       class Parallel < Jobs
+        def submit
+        end
+
         def wait_for_completion(verbose)
+          @jobs.each do |job|
+            job.submit
+          end
```

Comment:
I have a feeling this should call the submit method here, and that the submit method should actually submit all the jobs. Similarly, `Sequence` should probably submit the first job from the submit method. Otherwise, I have a feeling that:

```ruby
parallel do
  sequence { ... }
  sequence { ... }
end
```
would still submit the first two jobs in the two sequences from concurrent threads. For the parallel case to actually matter, it would need to be something along the lines of

```ruby
parallel do
  parallel { ... }
  parallel { ... }
end
```

which doesn't make a lot of sense in isolation (as the parallel blocks could be inlined), but I see how that could happen in a scenario where the inner parallels are part of some abstraction.